### ISSUE TITLE
Add outline destructuring assignments.

### ISSUE THREAD

**Issue by KodyJKing** [glassorg/ionscript#3]
Possible syntax:
```
let {} = props
     zoomRate = 0.004, xAxis = true, style, navState
     width, height = 200
     minViewWidth, maxViewWidth
     viewY, viewHeight
     ...otherProps
```
Multiple assignments on one line is consistent with https://github.com/glassorg/ionscript/issues/1

**Comment by krisnye** [comment]
What about this?

Possible syntax:
```
let {}
    zoomRate = 0.004, xAxis = true, style, navState
    ...otherProps
= props

// which also works for
{}
    zoomRate = 0.004, xAxis = true, style, navState
    ...otherProps
= props

```

**Comment by KodyJKing** [comment]
That works for me.

**Comment by krisnye** [comment]
{
        foo, bar
        ...rest
    } = value

    [
        foo, bar
        {
            a, b
            ...more
        }
        ...rest
    ] = value

**Comment by krisnye** [comment]
It's weird because it's the only syntax which is not cleanly outdented. Probably OK. We are consistently outdented for all "construction" of objects. This syntax is only for "destruction" of objects.

**Comment by krisnye** [comment]
Fixed and published as 0.4.7

https://www.npmjs.com/package/ionscript
